src/Lykke.Service.SmsSender/Controllers/CallbackController.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System.Threading.Tasks;
+using Common.Log;
+using Lykke.Cqrs;
+using Lykke.Service.SmsSender.Core.Domain.SmsRepository;
+using Lykke.Service.SmsSender.Models;
+using Lykke.Service.SmsSender.Sagas.Commands;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Lykke.Service.SmsSender.Controllers
+{
+    [Route("callback")]
+    [ApiExplorerSettings(IgnoreApi=true)]
+    public class CallbackController : Controller
+    {
+        private readonly ISmsRepository _smsRepository;
+        private readonly ICqrsEngine _cqrsEngine;
+        private readonly ILog _log;
+
+        public CallbackController(
+            ISmsRepository smsRepository,
+            ICqrsEngine cqrsEngine,
+            ILog log
+        )
+        {
+            _smsRepository = smsRepository;
+            _cqrsEngine = cqrsEngine;
+            _log = log;
+        }
+
+        [HttpPost]
+        [Route("twilio")]
+        public async Task<IActionResult> TwilioCallback([FromForm] TwilioCallbackModel model)
+        {
+            if (model == null)
+                return Ok();
+            
+            _log.WriteInfo(nameof(TwilioCallback), model, "Twilio callback");
+            
+            if (!string.IsNullOrEmpty(model.MessageSid))
+            {
+                var sms = await _smsRepository.GetByMessageIdAsync(model.MessageSid);
+
+                if (sms == null)
+                {
+                    _log.WriteWarning(nameof(TwilioCallback), model, $"Sms message with messageId = {model.MessageSid} not found");
+                    return Ok();
+                }
+                
+                switch (model.MessageStatus)
+                {
+                    case TwilioMessageStatus.Sent:
+                    case TwilioMessageStatus.Delivered:
+                        _cqrsEngine.SendCommand(new SmsDeliveredCommand {Message = sms}, "sms", "sms");
+                        break;
+                    case TwilioMessageStatus.Failed:
+                    case TwilioMessageStatus.Undelivered:
+                        _cqrsEngine.SendCommand(new SmsNotDeliveredCommand {Message = sms, Error = $"status = {model.MessageStatus}"}, "sms", "sms");
+                        break;
+                    default:
+                        _log.WriteWarning(nameof(TwilioCallback), model, $"status = {model.MessageStatus}, callback processing is skipped");
+                        break;
+                }
+            }
+
+            return Ok();
+        }
+
+        [HttpPost]
+        [HttpGet]
+        [Route("nexmo")]
+        public async Task<IActionResult> NexmoCallback(NexmoCallbackModel model)
+        {
+            if (model == null)
+                return Ok();
+            
+            _log.WriteInfo(nameof(NexmoCallback), model, "Nexmo callback");
```

Comment:
we have this log on error
_log.WriteWarning(nameof(SendSmsAsync), new
.Sid,
= response.To.SanitizePhone(),
.Status,
.ErrorCode,
.ErrorMessage
}, "twilio error response");

in context we have property Phone
here log about success responce, will be good add in context property Phone
abd after we can simple make visualization in kibana - success\error by phone codes

Reply:
the same as for Twilio